api/envoy/config/route/v3/route_components.proto

```diff
@@ -760,10 +760,84 @@ message RouteAction {
     ConnectConfig connect_config = 3;
   }
 
+  // Indicates that some headers received from upstream cluster need to be rewritten. When upstream
+  // returns a header, the header will be overwritten with the given value before being sent back as
+  // response to the client.
+  //
+  // In the below example there are two response header rewrites mentioned: one for location header
+  // and one for x-random-token header.
+  //
+  // In the location header's case, three things need to match before a value substitution happens for
+  // the location header - Location value prefix, Other headers present in response, and status code returned.
+  //
+  // If the value of the location header sent from upstream begins with /itm/not-found, first condition is met.
+  // If the upstream response has given values for *both* content-type and cookie headers, second condition is met.
+  // The response_codes condition can be met if *any* of the given response codes is returned from upstream.
+  // The string match section can use prefix match and present match only. In case of prefix match, the matched
+  // part is substituted. In case of present match the whole value of the header is overwritten.
+  //
+  // The x-random-token header's value will be matched using the regex pattern specified in the pattern
+  // section, and the matched part will be replaced by the value provided in the substitution section.
```

Comment:
Can you comment individual fields rather than a large detached block comment? Please take a look at the generated docs to see how they render, e.g. https://380489-65214191-gh.circle-artifacts.com/0/generated/docs/api-v3/config/route/v3/route_components.proto.html#envoy-v3-api-msg-config-route-v3-routeaction-headerrewriter